test/functional/tools/collection_creates_dynamic_nested.xml

```diff
@@ -0,0 +1,61 @@
+<tool id="collection_creates_dynamic_nested" name="collection_creates_dynamic_nested" version="0.1.0">
+  <command>
+    echo "A" > oe1_ie1.fq ;
+    echo "B" > oe1_ie2.fq ;
+    echo "C" > oe2_ie1.fq ;
+    echo "D" > oe2_ie2.fq ;
+    echo "E" > oe3_ie1.fq ;
+    echo "F" > oe3_ie2.fq
+  </command>
+  <inputs>
+    <param name="foo" type="text" label="Dummy Parameter" />
+  </inputs>
+  <outputs>
+    <collection name="list_output" type="list:list" label="Duplicate List">
+      <discover_datasets pattern="(?P&lt;identifier_0&gt;[^_]+)_(?P&lt;identifier_1&gt;[^_]+)\.fq" ext="fastqsanger" visible="true" />
```

Comment:
Since this serves often as documentation, can we document this scary line a little bit more. Also for the other examples.